models/articles.models.js

```diff
@@ -1,20 +1,31 @@
 const db = require("../db/connection");
 
 exports.selectArticlesById = async (article_id) => {
-  const comment = await db.query(
-    `SELECT COUNT(comment_id) AS count FROM comments WHERE comments.article_id = $1;`,
-    [article_id]
-  );
-  const comment_count = Number(comment.rows[0].count);
-  const result = await db.query(
-    `SELECT * FROM articles WHERE article_id = $1;`,
-    [article_id]
-  );
+  let articleQuery = `SELECT * FROM articles`;
+  let result;
+
+  if (!article_id) {
+    articleQuery += ` ORDER BY created_at DESC`;
+    result = await db.query(articleQuery);
+  } else {
+    articleQuery += ` WHERE article_id = $1;`;
+    result = await db.query(articleQuery, [article_id]);
+  }
+
+  for (const row of result.rows) {
+    let id = row.article_id;
+    const comment = await db.query(
+      `SELECT COUNT(comment_id) AS count FROM comments WHERE comments.article_id = $1;`,
+      [id]
+    );
+    const comment_count = Number(comment.rows[0].count);
+    row.comment_count = comment_count;
+  }
+
   if (result.rows.length === 0) {
     return Promise.reject({ status: 404, msg: "Article not found" });
   } else {
-    result.rows[0].comment_count = comment_count;
-    return result.rows[0];
+    return result.rows;
```

Comment:
Amazing that you've managed to handle all conditions for both get ALL articles and get articles by ID in one model! 🎉